lib/active_merchant/billing/gateways/cyber_source_rest.rb

```diff
@@ -398,21 +398,21 @@ def success_from(response)
       def message_from(response)
         return response['status'] if success_from(response)
 
-        response['errorInformation']['message'] || response['message']
+        response.dig('errorInformation', 'message') || response['message']
       end
 
       def authorization_from(response)
         id = response['id']
-        has_amount = response['orderInformation'] && response['orderInformation']['amountDetails'] && response['orderInformation']['amountDetails']['authorizedAmount']
-        amount = response['orderInformation']['amountDetails']['authorizedAmount'].delete('.') if has_amount
+        has_amount = response['orderInformation'] && response.dig('orderInformation', 'amountDetails') && response.dig('orderInformation', 'amountDetails', 'authorizedAmount')
```

Comment:
dig makes this much less complex
```suggestion
        has_amount = response.dig('orderInformation', 'amountDetails', 'authorizedAmount')
```

Reply:
Sounds good! I wonder why it was even looking for amountDetails